pyvelox/pyvelox.h

```diff
@@ -504,6 +527,56 @@ static void addVectorBindings(
             std::move(baseVector),
             PyVeloxContext::getSingletonInstance().pool());
       });
+
+  m.def(
+      "row_vector",
+      [](std::vector<std::string>& names,
+         std::vector<VectorPtr>& values,
+         const std::optional<py::list>& nullability_list) {
+        std::vector<std::shared_ptr<const Type>> childTypes;
+        childTypes.resize(values.size());
+        for (int i = 0; i < values.size(); i++) {
+          childTypes[i] = values[i]->type();
+        }
+        auto rowType = ROW(std::move(names), std::move(childTypes));
+        const size_t vectorSize = values.size();
+        BufferPtr nullability_buffer = nullptr;
+        if (nullability_list.has_value()) {
+          auto nullability_values = nullability_list.value();
+          nullability_buffer = AlignedBuffer::allocate<bool>(
+              nullability_values.size(),
+              PyVeloxContext::getSingletonInstance().pool());
+          for (size_t idx = 0; idx < nullability_values.size(); idx++) {
+            if (!py::isinstance<py::bool_>(nullability_values[idx])) {
+              throw py::type_error("Found a value that's not a bool");
+            }
+            bool value = py::cast<bool>(nullability_values[idx]);
+            bits::setBit(
+                nullability_buffer->asMutable<uint64_t>(),
+                idx,
+                bits::kNull ? value : !value);
+          }
+        }
+
+        return std::make_shared<RowVector>(
+            PyVeloxContext::getSingletonInstance().pool(),
+            rowType,
+            nullability_buffer,
+            vectorSize,
```

Comment:
Ideally we should take the size of the vector from the user (maybe optional argument) , else compute it as smallest child size if the child sizes are different ? But I feel  we should be explicit - I prefer throwing an error if children are different sizes.

Reply:
Good catch. I will add the changes.